src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.OpenSslAsync.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            (PipeStream client, PipeStream server) = ConcurrentObjectProvider.CreatePipePair();
+            await DefaultHandshake(client, server);
+            client.Dispose();
+            server.Dispose();
+        });
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task ConcurrentDefaultHandshakeContextIPv4Async() => Spawn(ConcurrentTasks, async () =>
+        {
+            (NetworkStream client, NetworkStream server) = ConcurrentObjectProvider.CreateIPv4Pair();
+            await DefaultContextHandshake(client, server);
+            client.Dispose();
+            server.Dispose();
+        });
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task ConcurrentDefaultHandshakeContextIPv6Async() => Spawn(ConcurrentTasks, async () =>
+        {
+            (NetworkStream client, NetworkStream server) = ConcurrentObjectProvider.CreateIPv6Pair();
+            await DefaultContextHandshake(client, server);
+            client.Dispose();
+            server.Dispose();
+        });
+        
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        [ArgumentsSource(nameof(TlsProtocols))]
+        public Task ConcurrentHandshakeContosoAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._cert, protocol));
+
+        [Benchmark]
+        [ArgumentsSource(nameof(TlsProtocols))]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task ConcurrentHandshakeECDSA256CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._ec256Cert, protocol));
+
+        [Benchmark]
+        [ArgumentsSource(nameof(TlsProtocols))]
+        [BenchmarkCategory(Categories.NoAOT)]
+        [OperatingSystemsFilter(allowed: true, platforms: OS.Linux)]    // Not supported on Windows at the moment.
+        public Task ConcurrentHandshakeECDSA512CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._ec512Cert, protocol));
+
+        [Benchmark]
+        [ArgumentsSource(nameof(TlsProtocols))]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task ConcurrentHandshakeRSA2048CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._rsa2048Cert, protocol));
+
+        [Benchmark]
+        [ArgumentsSource(nameof(TlsProtocols))]
+        public Task ConcurrentHandshakeRSA4096CertAsync(SslProtocols protocol) => Spawn(ConcurrentTasks, async () => await HandshakeAsync(SslStreamTests._rsa4096Cert, protocol));
+
+        private static async Task Spawn(int count, Func<Task> method)
+        {
+            var _tasks = new Collections.Generic.List<Task>(count);
+
+            for(int i = 0; i < count; ++i)
+            {
+                _tasks.Add(Task.Run(method));
+            }
+
+            await Task.WhenAll(_tasks);
+        }
+    }
+
```

Comment:
I would do something like this....

```C#
private static void DoBench(int taskCount, Func<Task> method)
{
    for (int i = 0; i < topLevelIters; i++)
    {
           var _tasks = new Collections.Generic.List<Task>(taskCount);

            for(int j = 0; j < taskCount; ++j)
            {
                _tasks.Add(Task.Run(method));
            }

            await Task.WhenAll(_tasks);
    }
}
```

with `taskCount` = 100 and `topLevelIters` = 10, that would be the 10,000 total tasks, 100 concurrent tasks of your previous micro. See my comments above.

In addition, what you have written does not guarantee the tasks will be executed concurrently, only that they _may_ be.